Make Ogg support optionl so that 'ogg' flag can be used to specify dependency
on libogg.

See upstream:
http://sourceforge.net/tracker/index.php?func=detail&aid=1271624&group_id=13478&atid=313478

Index: flac-1.1.2/configure.in
===================================================================
--- flac-1.1.2.orig/configure.in
+++ flac-1.1.2/configure.in
@@ -151,12 +151,26 @@ AC_DEFINE(FLAC__VALGRIND_TESTING)
 AM_CONDITIONAL(FLaC__HAS_XMMS, test -n "$XMMS_INPUT_PLUGIN_DIR")
 
 dnl check for ogg library
-XIPH_PATH_OGG(have_ogg=yes, AC_MSG_WARN([*** Ogg development enviroment not installed - Ogg support will not be built]))
-AM_CONDITIONAL(FLaC__HAS_OGG, [test "x$have_ogg" = xyes])
-if test "x$have_ogg" = xyes ; then
-AC_DEFINE(FLAC__HAS_OGG)
-AH_TEMPLATE(FLAC__HAS_OGG, [define if you have the ogg library])
+AC_ARG_ENABLE([ogg],
+	      AC_HELP_STRING([--disable-ogg], [Disable ogg support (default:test)]),
+	      [want_ogg=$enableval ], [ want_ogg=yes ] )
+
+AM_CONDITIONAL(FLaC__HAS_OGG, test x$want_ogg = xyes)
+
+if test "x$want_ogg" = "xyes"; then
+	PKG_CHECK_MODULES([OGG], [ogg], [have_ogg=yes], [have_ogg=no])
+
+	if test "x$have_ogg" = "xyes"; then
+		AC_DEFINE(FLAC__HAS_OGG)
+		AH_TEMPLATE(FLAC__HAS_OGG, [define if you have the ogg library])
+	else
+		AC_MSG_WARN(*** Ogg headers not found.  Ogg support disabled.)
+	fi
+else
+	AC_MSG_RESULT([checking for ogg support... disabled])
 fi
+AC_SUBST(OGG_CFLAGS)
+AC_SUBST(OGG_LIBS)
 
 dnl check for i18n(internationalization); these are from libiconv/gettext
 AM_ICONV
